### PR TITLE
util: stop no tracer

### PR DESCRIPTION
**Pull request by andreimatei**
Before this patch, a Stopper would create a Tracer if it was not
configured with one. This was a bad idea because:
1) In production, we ended up construction a Tracer for no reason, only
   to be overridden later with the Server's constructor.
2) A lot of tests create Stoppers, and use them to create subsystems
   which they then inject in various ways (generally fairly hacky) into
   a Server which has its own, different, Tracer. The two Tracers then
   interact in subtle ways in Traces (the stopper's tracer ends up creating
   either a parent or a child for spans coming from the server's tracer).
   That's not going to be allowed any more because it's pretty broken.

This patch makes the Stopper's use of tracing optional. If the Stopper
is not configured with a Tracer, it simply won't produce any spans. This
solves the mess with tests, and it also seems sane as far as the Stopper
is concerned as a stand-alone library.
The patch is technically stricter than it needs to be - a Stopper
without a Tracer could still create child spans using the parent's
Tracer. But it doesn't seem worth the code to do so for the benefit of
random tests that don't care anyway.

Release note: None

### CODE REVIEW

**Comment by andreimatei** [comment]
hold off on this please; I sent it out too early.

**Review by knz — APPROVED** [review]

**Comment by knz** [comment]
oh oops

lmk when I should review again

**Comment by andreimatei** [comment]
Thanks for the look, but I'm gonna close this cause it was broken. It's important for the stopper to always create a span for async tasks because otherwise the parent's span can leak into the closure and be used after it was finished. And even when there's no parent span, it's still a good idea to create a span for the task because different code expects it. We've been pretty inconsistent on protecting against `tracing.SpanFromContext()` returning nil.
So instead of this, I'm gonna try to fix all the tests that mix tracers in dubious ways. For a start, I've sent you #72924.

**Pull request closed without merging**
